### PR TITLE
Object state management for v2.1.0

### PR DESCRIPTION
**Pull request by phalasz**

### CODE REVIEW

File: API/Collection/ObjectStateCollection.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Kaliop\eZMigrationBundle\API\Collection;
+
+/**
+ * @todo add phpdoc to suggest typehinting
+ */
+class ObjectStateCollection extends AbstractCollection
+{
+    protected $allowedClass = 'eZ\Publish\API\Repository\Values\ObjectState\ObjectState';
```
Comment (mareg) [inline API/Collection/ObjectStateCollection.php]: For better usability, I'd use `::class` to get the right class, as it allows click through:

``` php
protected $allowedClass = \eZ\Publish\API\Repository\Values\ObjectState\ObjectState::class;
```

(unless some specific eZ things prevent from it)

File: API/Collection/ObjectStateCollection.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+}
+
```
Comment (mareg) [inline API/Collection/ObjectStateCollection.php]: remove empty line :trollface:

File: API/Collection/ObjectStateGroupCollection.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+namespace Kaliop\eZMigrationBundle\API\Collection;
+
+/**
+ * @todo add phpdoc to suggest typehinting
+ */
+class ObjectStateGroupCollection extends AbstractCollection
+{
+    protected $allowedClass = 'eZ\Publish\API\Repository\Values\ObjectState\ObjectStateGroup';
+}
+
```
Comment (mareg) [inline API/Collection/ObjectStateGroupCollection.php]: remove empty line :trollface:

File: API/Collection/ObjectStateGroupCollection.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Kaliop\eZMigrationBundle\API\Collection;
+
+/**
+ * @todo add phpdoc to suggest typehinting
+ */
+class ObjectStateGroupCollection extends AbstractCollection
+{
+    protected $allowedClass = 'eZ\Publish\API\Repository\Values\ObjectState\ObjectStateGroup';
```
Comment (mareg) [inline API/Collection/ObjectStateGroupCollection.php]: Same here, use `::class` (if possible)

File: Core/Executor/ObjectStateManager.php [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        $objectStateCreateStruct->defaultLanguageCode = self::DEFAULT_LANGUAGE_CODE;
+
+        foreach ($this->dsl['names'] as $name) {
+            if ( array_key_exists('languageCode', $name) && array_key_exists('name', $name)) {
+                $objectStateCreateStruct->names[$name['languageCode']] = $name['name'];
+            }
+        }
+
+        if (0 === count($objectStateCreateStruct->names)) {
+            throw new \Exception('No object state names have been defined. Need to specify atleast one to create the state.');
```
Comment (mareg) [inline Core/Executor/ObjectStateManager.php]: `atleast` => `at least`

**Review by mareg — APPROVED** [review]

File: API/Collection/ObjectStateCollection.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+namespace Kaliop\eZMigrationBundle\API\Collection;
+
+use eZ\Publish\API\Repository\Values\ObjectState\ObjectState;
+
+/**
+ * @todo add phpdoc to suggest typehinting
+ */
+class ObjectStateCollection extends AbstractCollection
+{
+    protected $allowedClass = ObjectState::class;
```
Comment (andrerom) [inline API/Collection/ObjectStateCollection.php:12]: composer.json should bump php requirements to 5.5 for this one

**Pull request merged**